client/src/components/AddCard/AddCardInfo/AddCardInfoForm/AddCardInfoForm.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import { useStripe, useElements, PaymentElement } from '@stripe/react-stripe-js';
+
+const SetupForm = () => {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleSubmit = async (event: { preventDefault: () => void }) => {
+    // We don't want to let default form submission happen here,
+    // which would refresh the page.
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      // Stripe.js has not yet loaded.
+      // Make sure to disable form submission until Stripe.js has loaded.
+      return;
+    }
+
+    const { error } = await stripe.confirmSetup({
+      //`Elements` instance that was used to create the Payment Element
+      elements,
+      confirmParams: {
+        return_url: process.env.REACT_APP_DOMAIN + '/dashboard',
+      },
+    });
+
+    if (error.message) {
+      // This point will only be reached if there is an immediate error when
+      // confirming the payment. Show error to your customer (e.g., payment
+      // details incomplete)
+      setErrorMessage(error.message);
+    } else {
+      // Your customer will be redirected to your `return_url`. For some payment
+      // methods like iDEAL, your customer will be redirected to an intermediate
+      // site first to authorize the payment, then redirected to the `return_url`.
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <PaymentElement />
+      <button disabled={!stripe}>Submit</button>
+      {/* Show error message to your customers */}
```

Comment:
This comment is not needed. Can we remove it, please?

Reply:
👍Done